test/common/stats/stats_matcher_test.cc

```diff
@@ -0,0 +1,245 @@
+#include "envoy/config/metrics/v2/stats.pb.h"
+
+#include "common/stats/stats_matcher_impl.h"
+
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::IsFalse;
+using testing::IsTrue;
+
+namespace Envoy {
+namespace Stats {
+
+class StatsMatcherTest : public testing::Test {
+protected:
+  envoy::type::matcher::StringMatcher* inclusionList() {
+    return stats_config_.mutable_stats_matcher()->mutable_inclusion_list()->add_patterns();
+  }
+  envoy::type::matcher::StringMatcher* exclusionList() {
+    return stats_config_.mutable_stats_matcher()->mutable_exclusion_list()->add_patterns();
+  }
+  void rejectAll(const bool should_reject) {
+    stats_config_.mutable_stats_matcher()->set_reject_all(should_reject);
+  }
+
+  void initMatcher() { stats_matcher_impl_ = std::make_unique<StatsMatcherImpl>(stats_config_); }
+
+  envoy::config::metrics::v2::StatsConfig stats_config_;
```

Comment:
nit: this should go at the end of its scope, below the other protected methods (but it's not used outside of this class so it could also go in the private section)

Reply:
Done.